src/jit.rs

```diff
@@ -1424,25 +1430,66 @@ impl<'a, C: ContextObject> JitCompiler<'a, C> {
             self.set_anchor(ANCHOR_TRANSLATE_MEMORY_ADDRESS + target_offset);
-            // call MemoryMapping::map() storing the result in RuntimeEnvironmentSlot::ProgramResult
-            self.emit_rust_call(Value::Constant64(MemoryMapping::map as *const u8 as i64, false), &[
-                Argument { index: 3, value: Value::Register(R11) }, // Specify first as the src register could be overwritten by other arguments
-                Argument { index: 5, value: Value::Constant64(0, false) }, // self.pc is set later
-                Argument { index: 4, value: Value::Constant64(*len as i64, false) },
-                Argument { index: 2, value: Value::Constant64(*access_type as i64, false) },
-                Argument { index: 1, value: Value::RegisterPlusConstant32(RBP, self.slot_on_environment_stack(RuntimeEnvironmentSlot::MemoryMapping), false) },
-                Argument { index: 0, value: Value::RegisterPlusConstant32(RBP, self.slot_on_environment_stack(RuntimeEnvironmentSlot::ProgramResult), false) },
-            ], None);
+            // call MemoryMapping::(map|load|store) storing the result in RuntimeEnvironmentSlot::ProgramResult
+            if *len == 1 {
+                self.emit_rust_call(Value::Constant64(MemoryMapping::map as *const u8 as i64, false), &[
+                    Argument { index: 3, value: Value::Register(R11) }, // Specify first as the src register could be overwritten by other arguments
+                    Argument { index: 5, value: Value::Constant64(0, false) }, // self.pc is set later
+                    Argument { index: 4, value: Value::Constant64(*len as i64, false) },
+                    Argument { index: 2, value: Value::Constant64(*access_type as i64, false) },
+                    Argument { index: 1, value: Value::RegisterPlusConstant32(RBP, self.slot_on_environment_stack(RuntimeEnvironmentSlot::MemoryMapping), false) },
+                    Argument { index: 0, value: Value::RegisterPlusConstant32(RBP, self.slot_on_environment_stack(RuntimeEnvironmentSlot::ProgramResult), false) },
+                ], None);
+            } else if *access_type == AccessType::Load {
+                let load = match len {
+                    1 => MemoryMapping::load::<u8> as *const u8 as i64,
+                    2 => MemoryMapping::load::<u16> as *const u8 as i64,
+                    4 => MemoryMapping::load::<u32> as *const u8 as i64,
+                    8 => MemoryMapping::load::<u64> as *const u8 as i64,
+                    _ => unreachable!()
+
+                };
+                self.emit_rust_call(Value::Constant64(load, false), &[
+                    Argument { index: 2, value: Value::Register(R11) }, // Specify first as the src register could be overwritten by other arguments
+                    Argument { index: 4, value: Value::Constant64(0, false) }, // self.pc is set later
+                    Argument { index: 3, value: Value::Constant64(*len as i64, false) },
```

Comment:
In the signature there is no `len` parameter anymore:
`pub fn load<T: Pod + Into<u64>>(&self, vm_addr: u64, pc: usize) -> ProgramResult`